shared/reducers/chat.js

```diff
@@ -26,19 +26,54 @@ function reducer (state: State = initialState, action: Actions) {
             .set('isLoading', false)
         })
 
-      return state.set('conversationStates', newConversationStates)
+      // Reset the unread count
+      const newInboxStates = state.get('inbox').map(inbox => inbox.get('conversationIDKey') !== conversationIDKey ? inbox : inbox.set('unreadCount', 0))
+
+      return state
+        .set('conversationStates', newConversationStates)
+        .set('inbox', newInboxStates)
     }
     case Constants.appendMessages: {
-      const appendMessages = action.payload.messages
+      const appendAction: AppendMessages = action
+      const appendMessages = appendAction.payload.messages
+      const message: Message = appendMessages[0]
+      const conversationIDKey = appendAction.payload.conversationIDKey
       const newConversationStates = state.get('conversationStates').update(
-        action.payload.conversationIDKey,
+        conversationIDKey,
         initialConversation,
         conversation => conversation.set('messages', conversation.get('messages').push(...appendMessages)))
 
-      return state.set('conversationStates', newConversationStates)
+      const isSelected = state.get('selectedConversation') === action.payload.conversationIDKey
+      let snippet
+
+      switch (message.type) {
+        case 'Text':
+          snippet = makeSnippet(message.message && message.message.stringValue(), 100)
+          break
+        default:
+          snippet = ''
+      }
+
+      const newInboxStates = state.get('inbox').map(inbox => (
+        inbox.get('conversationIDKey') !== conversationIDKey
+          ? inbox
+          : inbox
+            .set('unreadCount', isSelected ? 0 : inbox.get('unreadCount') + appendMessages.length)
```

Comment:
It seems like you are modifying the unreadCount in three different actions. I thought we wanted more orthogonal actions that modify a single spot in the store. In that case it might make sense to have a separate action to modify the unread count and that logic would live in the actions